Source/RageCommunity.Library/Extensions/VectorExtensions.cs

```diff
@@ -62,5 +62,44 @@ public static Vector3 GetOffset(this Vector3 from, float heading, Vector3 offset
 
             return new Vector3(from.X + resultX, from.Y + resultY, from.Z + offset.Z);
         }
+        /// <summary>
+        /// get heading toward a position
+        /// </summary>
+        /// <param name="from"></param>
+        /// <param name="otherVector">the <see cref="Vector3"/> to face to</param>
+        /// <returns>the heading toward a position</returns>
+        /// <remarks>
+        /// Source: <a href="http://answers.unity.com/answers/697834/view.html"></a>
+        /// </remarks>
+        public static float GetHeadingTowardsVector(this Vector3 from, Vector3 otherVector)
+        {
+            var direction = from - otherVector;
+            return MathHelper.ConvertDirectionToHeading(direction.ToNormalized());
+        }
+        /// <summary>
+        /// get heading towards an entity
+        /// </summary>
+        /// <param name="from"></param>
+        /// <param name="entity">the <see cref="Entity"/> to face to</param>
+        /// <returns>the heading towards the given <paramref name="entity"/></returns>
+        public static float GetHeadingTowardsEntity(this Vector3 from, Entity entity) => GetHeadingTowardsVector(from, entity.Position);
+        /// <summary>
+        /// Determine whether the given <paramref name="position"/> is on screen, or visible by a rendering <see cref="Camera"/>
```

Comment:
Is this summary accurate for the "IsOccupied" methods?

Reply:
Opps, my bad